src/ray/raylet/scheduling/cluster_resource_manager.cc

```diff
@@ -282,24 +199,8 @@ bool ClusterResourceManager::AddNodeAvailableResources(
   }
 
   auto node_resources = it->second.GetMutableLocalView();
-  RAY_CHECK(resource_request.predefined_resources.size() <=
-            node_resources->predefined_resources.size());
-
-  for (size_t i = 0; i < resource_request.predefined_resources.size(); ++i) {
-    node_resources->predefined_resources[i].available +=
-        resource_request.predefined_resources[i];
-    node_resources->predefined_resources[i].available =
-        std::min(node_resources->predefined_resources[i].available,
-                 node_resources->predefined_resources[i].total);
-  }
-  for (auto &entry : resource_request.custom_resources) {
-    auto it = node_resources->custom_resources.find(entry.first);
-    if (it != node_resources->custom_resources.end()) {
-      it->second.available += entry.second;
-      it->second.available = std::min(it->second.available, it->second.total);
-    }
-  }
-
+  node_resources->available += resource_request;
```

Comment:
Can we check to make sure resource_request doesn't contain custom resources that node_resources doesn't have? Otherwise I think the old code and the new one don't have the same behavior.

Reply:
Good catch. 

@scv119 what do you think is the correct semantics of this function? Should it only add existing resources?

There is another "UpdateNodeAvailableResourcesIfExist" function that explicitly says it only update existing resources. But this "AddNodeAvailableResources" function doesn't.

Reply:
I'll keep the behavior unchanged in this PR (only add resources that only exist in node_resources).

Reply:
yeah better keep the current behavior... probably we should merge it with UpdateNodeAvailableResourcesIfExist in follow up PRs.